core/services/blockhashstore/coordinators.go

```diff
@@ -0,0 +1,171 @@
+package blockhashstore
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/pkg/errors"
+
+	v1 "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/solidity_vrf_coordinator_interface"
+	v2 "github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/vrf_coordinator_v2"
+)
+
+var (
+	_ Coordinator = MultiCoordinator{}
+	_ Coordinator = &V1Coordinator{}
+	_ Coordinator = &V2Coordinator{}
+)
+
+// MultiCoordinator combines the data from multiple coordinators.
+type MultiCoordinator []Coordinator
+
+// NewMultiCoordinator creates a new Coordinator that combines the results of the given
+// coordinators.
+func NewMultiCoordinator(coordinators ...Coordinator) Coordinator {
+	if len(coordinators) == 1 {
+		return coordinators[0]
+	}
+	return MultiCoordinator(coordinators)
+}
+
+// Requests satisfies the Coordinator interface.
+func (m MultiCoordinator) Requests(
+	ctx context.Context,
+	fromBlock uint64,
+	toBlock uint64,
+) ([]Event, error) {
+	var reqs []Event
+	for _, c := range m {
+		r, err := c.Requests(ctx, fromBlock, toBlock)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
+		}
+		reqs = append(reqs, r...)
+	}
+	return reqs, nil
+}
+
+// Fulfillments satisfies the Coordinator interface.
+func (m MultiCoordinator) Fulfillments(ctx context.Context, fromBlock uint64) ([]Event, error) {
+	var fuls []Event
+	for _, c := range m {
+		f, err := c.Fulfillments(ctx, fromBlock)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
+		}
+		fuls = append(fuls, f...)
+	}
+	return fuls, nil
+}
+
+// V1Coordinator fetches request and fulfillment logs from a VRF V1 coordinator contract.
+type V1Coordinator struct {
+	c v1.VRFCoordinatorInterface
```

Comment:
I _think_ you can just embed this member by not giving it a member name:

```golang
type V1Coordinator struct {
  v1.VRFCoordinatorInterface
}
```

then you can call methods like:

```golang
iter, err := v.FilterRandomnessRequest(...)
```